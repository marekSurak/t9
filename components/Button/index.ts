import { styled } from 'styles/stitches.config'

export const Button = styled('button', {
  display: 'flex',
  flexFlow: 'column',
  justifyContent: 'center',
  alignItems: 'center',
  background: '$lightGray',

  border: 'none',
  borderRadius: '50%',
  padding: '$3',
  width: '$5',
  height: '$5',

  '&:hover': {
    background: '$dimGray',
  },
})
