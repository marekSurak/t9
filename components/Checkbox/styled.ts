import { styled } from 'styles/stitches.config'

export const CheckboxContainer = styled('section', {
  height: '$2',
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'space-between',
  fontSize: '$3',
  backgroundColor: '$light',
  borderRadius: '$1',
  padding: '0 $3',
})
