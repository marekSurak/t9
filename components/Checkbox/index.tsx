import type { FC } from 'react'
import React from 'react'

import { CheckboxContainer } from './styled'

interface IProps {
  checked: boolean
  onClick: () => void
}

export const CHECKBOX_ID = 'checkbox'

const Checkbox: FC<IProps> = ({ checked, onClick }) => (
  <CheckboxContainer>
    <label htmlFor={CHECKBOX_ID}>Use word list</label>
    <input
      id={CHECKBOX_ID}
      type="checkbox"
      checked={checked}
      onChange={onClick}
    />
  </CheckboxContainer>
)

export const MemoizedCheckbox = React.memo(Checkbox)
