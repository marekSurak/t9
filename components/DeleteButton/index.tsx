import type { FC } from 'react'

import { Button, ButtonContainer } from './styled'

interface IProps {
  onClick: () => void
  children: React.ReactNode
}

export const DeleteButton: FC<IProps> = ({ onClick, children }) => (
  <ButtonContainer>
    <Button type="button" onClick={onClick}>
      {children}
    </Button>
  </ButtonContainer>
)
