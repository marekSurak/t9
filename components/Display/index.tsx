import type { FC } from 'react'

import { DisplayContainer } from './styled'

interface IProps {
  children: React.ReactNode
}

export const Display: FC<IProps> = ({ children }) => (
  <DisplayContainer>{children}</DisplayContainer>
)
