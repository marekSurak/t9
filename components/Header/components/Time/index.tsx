import type { FC } from 'react'
import { useCallback, useEffect, useState } from 'react'

import { TimeContainer } from './styled'

export const Time: FC = () => {
  const [time, setTime] = useState<string>('')

  const updateTime = useCallback(() => {
    const date = new Date()
    const hours = date.getHours()
    const minutes = date.getMinutes()

    setTime(`${hours}:${minutes}`)
  }, [setTime])

  useEffect(() => {
    updateTime()

    const interval = setInterval(() => updateTime(), 60000)

    return () => {
      clearInterval(interval)
    }
  }, [updateTime])

  return <TimeContainer>{time}</TimeContainer>
}
