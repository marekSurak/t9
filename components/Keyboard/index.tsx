import type { FC } from 'react'
import React from 'react'

import { Button } from 'components/Button'
import { keyboardButtons } from 'constants/keyboard'

import { ButtonNumber, ButtonText, KeyboardContainer } from './styled'

interface IProps {
  onClick: (value: number) => void
}

const Keyboard: FC<IProps> = ({ onClick }) => (
  <KeyboardContainer>
    {keyboardButtons.map((button) => (
      <Button
        key={button.value}
        type="button"
        onClick={() => onClick(button.value)}
      >
        <ButtonNumber>{button.value}</ButtonNumber>
        <ButtonText>{button.label}</ButtonText>
      </Button>
    ))}
  </KeyboardContainer>
)

export const MemoizedKeyboard = React.memo(Keyboard)
