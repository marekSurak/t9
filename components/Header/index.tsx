import type { FC } from 'react'
import React from 'react'

import { BatteryIcon } from 'components/icons/BatteryIcon'
import { SignalIcon } from 'components/icons/SignalIcon'
import { WifiIcon } from 'components/icons/WifiIcon'

import { Time } from './components/Time'
import { HeaderContainer, IconsContainer } from './styled'

const Header: FC = () => (
  <HeaderContainer>
    <Time />

    <IconsContainer>
      <SignalIcon />
      <WifiIcon />
      <BatteryIcon />
    </IconsContainer>
  </HeaderContainer>
)

export const MemiozedHeader = React.memo(Header)
