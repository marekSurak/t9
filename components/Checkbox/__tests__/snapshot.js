import { render } from '@testing-library/react'
import { MemoizedCheckbox as Checkbox } from 'components/Checkbox'

describe('Toggle', () => {
  it('Renders Toggle checked', () => {
    const { container } = render(<Checkbox checked={true} onClick={() => {}} />)
    expect(container).toMatchSnapshot()
  })

  it('Renders Toggle unchecked', () => {
    const { container } = render(
      <Checkbox checked={false} onClick={() => {}} />
    )
    expect(container).toMatchSnapshot()
  })
})
