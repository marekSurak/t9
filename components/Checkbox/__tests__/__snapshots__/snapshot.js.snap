// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toggle Renders Toggle checked 1`] = `
<div>
  <div
    class="c-igMNBK"
  >
    <label
      for="checkbox"
    >
      Use word list
    </label>
    <input
      checked=""
      id="checkbox"
      type="checkbox"
    />
  </div>
</div>
`;

exports[`Toggle Renders Toggle unchecked 1`] = `
<div>
  <div
    class="c-igMNBK"
  >
    <label
      for="checkbox"
    >
      Use word list
    </label>
    <input
      id="checkbox"
      type="checkbox"
    />
  </div>
</div>
`;
